### ISSUE TITLE
Duplicate key Layout/LineLength in `.rubocop_todo.yml` when Layout/LineLength is enabled via comment

### ISSUE THREAD

**Issue by guigs** [rubocop/rubocop#11916]
When generating `.rubocop_todo.yml` with `--auto-gen-only-exclude` option, if a file contains both a `Layout/LineLength` offense and disable `Layout/LineLength` rule for block (thus re-enabling it afterwards), then the resulting `.rubocop_todo.yml` contains two entries for `Layout/LineLength` and `rubocop` still reports offenses afterwards.

This issue don't happen in 1.45.1. It manifests from 1.46.0 until 1.52.0. I was able to bisect it to this commit: https://github.com/rubocop/rubocop/commit/efce268820e4dbbb4a85a61d3510ab351d4d0c4f (PR #11603).

## Expected behavior

* `.rubocop_todo.yml` should not have duplicate keys.
* running rubocop after generating `.rubocop_todo.yml` should not report any offenses.

## Actual behavior

* `.rubocop_todo.yml` contains duplicated blocks for `Layout/LineLength`, with different set of Exclude files.
* running rubocop after generating `.rubocop_todo.yml` still report offenses.

## Steps to reproduce the problem

Create an app with the following files:

`foo.rb`
```
# frozen_string_literal: true

# This is Foo
class Foo
  def foo
    puts 'This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo'
  end

  # rubocop:disable Layout/LineLength
  def bar
    puts 'This is Bar'
  end
  # rubocop:enable Layout/LineLength
end
```

`.rubocop.yml`

```
inherit_from: .rubocop_todo.yml

AllCops:
  NewCops: enable
```

Run 

```
rubocop --auto-gen-config --auto-gen-only-exclude
```

`.rubocop_todo.yml`

```
# This configuration was generated by
# `rubocop --auto-gen-config --auto-gen-only-exclude`
# on 2023-06-09 14:31:12 UTC using RuboCop version 1.52.0.
# The point is for the user to remove these configuration records
# one by one as the offenses are removed from the code base.
# Note that changes in the inspected code, or installation of new
# versions of RuboCop, may require this file to be generated again.

# Offense count: 1
# This cop supports safe autocorrection (--autocorrect).
# Configuration parameters: Max, AllowHeredoc, AllowURI, URISchemes, IgnoreCopDirectives, AllowedPatterns.
# URISchemes: http, https
Layout/LineLength:
  Exclude:
    - 'foo.rb'

# Offense count: 1
# This cop supports safe autocorrection (--autocorrect).
# Configuration parameters: Max, AllowHeredoc, AllowURI, URISchemes, IgnoreCopDirectives, AllowedPatterns.
# URISchemes: http, https
Layout/LineLength:
  Exclude:
    - 'foo.rb'
```

Run `rubocop`

```
.rubocop_todo.yml:13: `Layout/LineLength` is concealed by line 21
Inspecting 2 files
.C

Offenses:

foo.rb:6:121: C: Layout/LineLength: Line is too long. [130/120]
    puts 'This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo This is Foo'
                                                                                                                        ^^^^^^^^^^

2 files inspected, 1 offense detected
```

The problem does not happen with version 1.45.1. It happens from version 1.46.0 up to 1.52.0.

## Context

While it would be easy to workaround the examples above, our use case is that we use [Annotate gem](https://github.com/ctran/annotate_models) to generate schema info comments for our classes, and because some annotations lines can be longer than the defined LineLength, we wrap schema info annotations with `# rubocop:disable Layout/LineLength` and `# rubocop:enable Layout/LineLength`, using `wrapper_open` and `wrapper_close` options from `Annotate`.
But it also annotate specs and fabricators which we generally exclude from `Layout/LineLength` via an `Exclude` option `'spec/**/*'`.

The problems seems to be that `# rubocop:enable Layout/LineLength` actually enables the rule, even if file is in the exclusion list, instead of just restoring to the "previous state" it has before the `# rubocop:disable Layout/LineLength`.

**Comment by Earlopain** [comment]
Hi @guigs, could you check if this still is a problem for you? I tried reproducing but it seems to jus work for me.
